graspologic/pipeline/embed/omnibus_embedding.py

```diff
@@ -0,0 +1,280 @@
+            n_iter=svd_solver_iterations,
+            check_lcc=False,
+            diag_aug=False,
+            concat=False,
+            svd_seed=svd_seed,
+        )
+
+        previous_embedding, current_embedding = model.fit_transform(
+            graphs=[previous_graph_augmented, current_graph_augmented]
+        )
+
+        previous_embedding_cut = _elbow_cut_if_needed(
+            elbow_cut, graph.is_directed(), model.singular_values_, previous_embedding
+        )
+
+        current_embedding_cut = _elbow_cut_if_needed(
+            elbow_cut, graph.is_directed(), model.singular_values_, current_embedding
+        )
+
+        graph_embeddings.append(
+            (
+                Embeddings(union_node_ids, previous_embedding_cut),
+                Embeddings(union_node_ids, current_embedding_cut),
+            )
+        )
+
+    return graph_embeddings
+
+
+def _graphs_precondition_checks(graphs, weight_attribute):
+    is_directed = graphs[0].is_directed()
+
+    for graph in graphs:
+        check_argument(
+            is_directed == graph.is_directed(),
+            "graphs must either be all directed or all undirected",
+        )
+
+        check_argument(
+            not graph.is_multigraph(),
+            "Multigraphs are not supported; you must determine how to represent at most "
+            "one edge between any two nodes, and handle the corresponding weights "
+            "accordingly",
+        )
+
+        if not is_real_weighted(graph, weight_attribute=weight_attribute):
+            warnings.warn(
+                f"Graphs with edges that do not have a real numeric weight set for every "
+                f"{weight_attribute} attribute on every edge are treated as an unweighted "
+                f"graph - which presumes all weights are `1.0`. If this is incorrect, "
+                f"please add a '{weight_attribute}' attribute to every edge with a real, "
+                f"numeric value (e.g. an integer or a float) and call this function again."
+            )
+            weight_attribute = None  # this supercedes what the user said, because
+            # not all of the weights are real numbers, if they exist at all
+            # this weight=1.0 treatment actually happens in nx.to_scipy_sparse_matrix()
+
+    return weight_attribute
+
+
+def _elbow_cut_if_needed(elbow_cut, is_directed, singular_values, embedding):
+    if elbow_cut is None:
+        if is_directed:
+            embedding = np.concatenate(embedding, axis=1)
+    else:
+        column_index = _index_of_elbow(singular_values, elbow_cut)
+
+        if is_directed:
+            left, right = embedding
+            left = left[:, :column_index]
+            right = right[:, :column_index]
+            embedding = np.concatenate((left, right), axis=1)
+        else:
+            embedding = embedding[:, :column_index]
+
+    return embedding
+
+
+def _augment_graph(graph, node_ids, weight_attribute):
+    graph_as_array = nx.to_numpy_array(

```

Comment:
not sparse?

Reply:
Nope, when #621 pushes through I will make that change.